server.js

```diff
@@ -14,10 +21,81 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
+app.get("/", function (request, response)
+{
   response.sendFile(__dirname + "/index.html");
 });
 
 app.listen(process.env.PORT);
+
+
+
+
+
+app.get("/messages", (req, res) =>
+{
+  res.send(messages);
+});
+
+app.post("/messages", (req, res) =>
+{
+  let msgName = req.body.from;
+  let msgText = req.body.text;
+  let idPosition = messages.length;
+
+  const newMsg =
+  {
+    id: idPosition,
+    name: msgName,
+    message: msgText,
+    timeSent: new Date(),
+  };
+
+  if (!newMsg.name || !newMsg.message)
+  {
+    return res.status(400).json("Please include a name and message");
+  }
+
+  else
+  {
+    messages.push(newMsg);
+    res.status(200).json(messages);
+  }
+});
+
+
+
+app.get("/messages/:id", function (req, res)
+{
+  let id = parseInt(req.params.id);
+  let filterMsg = messages.filter(msg => msg.id === id);
+
+  res.send(filterMsg);
+});
+
+app.get("/messages/delete/:id", function (req, res)
+{
+  let id = parseInt(req.params.id);
+  let filterdMsg = messages.filter(msg => msg.id === id);
+
+  messages = messages.filter(msg => msg.id !== id);
+
```

Comment:
As above, checking the message exists and returning a 404 if it doesn't would be a nice improvement.